overwatch/src/services/life_cycle.rs

```diff
@@ -1,41 +1,80 @@
-use std::{default::Default, error::Error};
+use std::{
+    default::Default,
+    pin::Pin,
+    task::{Context, Poll},
+};
 
-use futures::Stream;
-use tokio::sync::broadcast::{channel, Receiver, Sender};
-use tokio_stream::StreamExt;
+use futures::{Stream, StreamExt};
+use tokio::sync::mpsc::{channel, Sender};
+use tokio_stream::wrappers::ReceiverStream;
 
-use crate::DynError;
+use crate::{utils::finished_signal, DynError};
 
-/// Type alias for an empty signal.
-pub type FinishedSignal = ();
+#[expect(clippy::derive_partial_eq_without_eq, reason = "Not needed")]
```

Comment:
I mean, ideally there is no downside to deriving `Eq` as there is no downside to derive `Debug`, so I'd rather see `Eq` derived than an `expect` here :)